subworkflows/local/bam_applybqsr/main.nf

```diff
@@ -4,44 +4,85 @@
 // A when clause condition is defined in the conf/modules.config to determine if the module should be run
 
-include { GATK4_APPLYBQSR           } from '../../../modules/nf-core/gatk4/applybqsr/main'
-include { CRAM_MERGE_INDEX_SAMTOOLS } from '../cram_merge_index_samtools/main'
+include { BAM_MERGE_INDEX_SAMTOOLS  } from '../bam_merge_index_samtools'
+include { CRAM_MERGE_INDEX_SAMTOOLS } from '../cram_merge_index_samtools'
+include { GATK4_APPLYBQSR           } from '../../../modules/nf-core/gatk4/applybqsr'
 
 workflow BAM_APPLYBQSR {
     take:
-    cram          // channel: [mandatory] [ meta, cram, crai, recal ]
-    dict          // channel: [mandatory] [ dict ]
-    fasta         // channel: [mandatory] [ fasta ]
-    fasta_fai     // channel: [mandatory] [ fasta_fai ]
-    intervals     // channel: [mandatory] [ intervals, num_intervals ] or [ [], 0 ] if no intervals
+    cram      // channel: [mandatory] [ meta, cram, crai, recal ]
+    dict      // channel: [mandatory] [ dict ]
+    fasta     // channel: [mandatory] [ fasta ]
+    fasta_fai // channel: [mandatory] [ fasta_fai ]
+    intervals // channel: [mandatory] [ intervals, num_intervals ] or [ [], 0 ] if no intervals
 
     main:
     versions = Channel.empty()
+    bam_applybqsr_single = Channel.empty()
+    bam_to_merge = Channel.empty()
 
     // Combine cram and intervals for spread and gather strategy
-    cram_intervals = cram.combine(intervals)
-        // Move num_intervals to meta map
-        .map{ meta, cram, crai, recal, intervals, num_intervals -> [ meta + [ num_intervals:num_intervals ], cram, crai, recal, intervals ] }
+    // Move num_intervals to meta map
+    cram_intervals = cram
+        .combine(intervals)
+        .map { meta, cram_, crai, recal, intervals_, num_intervals -> [meta + [num_intervals: num_intervals], cram_, crai, recal, intervals_] }
 
     // RUN APPLYBQSR
-    GATK4_APPLYBQSR(cram_intervals, fasta.map{ meta, it -> [ it ] }, fasta_fai.map{ meta, it -> [ it ] }, dict.map{ meta, it -> [ it ] })
+    GATK4_APPLYBQSR(
+        cram_intervals,
+        fasta.map { _meta, fasta_ -> [fasta_] },
+        fasta_fai.map { _meta, fasta_fai_ -> [fasta_fai_] },
+        dict.map { _meta, dict_ -> [dict_] },
+    )
+
+    // FOR BAMs
+    if (params.save_output_as_bam) {
+
+        bam_applybqsr_out = GATK4_APPLYBQSR.out.bam
+            .join(GATK4_APPLYBQSR.out.bai, failOnDuplicate: true, failOnMismatch: true)
+            .branch {
+                single: it[0].num_intervals == 1
+                multiple: it[0].num_intervals > 1
+            }
+
+        bam_applybqsr_single = bam_applybqsr_out.single
+
+        // For multiple intervals, gather and merge the recalibrated cram files
+        bam_to_merge = bam_applybqsr_out.multiple
+            .map { meta, bam_, _bai -> [groupKey(meta, meta.num_intervals), bam_] }
+            .groupTuple()
+    }
+
+    // Merge and index the recalibrated cram files
+    BAM_MERGE_INDEX_SAMTOOLS(bam_to_merge)
```

Comment:
bam_to_merge doesn't seem to be set for the cram output

Reply:
ah seems to be a typo in the comment

Reply:
can all these bam operations not move into the if block? It should keep the terminal output cleaner

Reply:
smart idea

Reply:
Actually no need for that, the operations are just on the channels, no process are in the if block, so we're good